### DISCUSSION TITLE
More granular scope of uniqueness?

### DISCUSSION THREAD

**Discussion by hunghvu** [payloadcms/payload#2678] [Feature Requests & Ideas]
At the moment, it appears `unique: true` is only collection-level. What if we,

1. Allow uniqueness at `array` level?
2. Allow merging multiple fields to form a unique key (composite keys)?

Feel free to let me know if there is a work around already, or how the proposal can be improved. Thank you.

**Comment by yuanpli** [comment]
This may can help. See #2529

**Reply by hunghvu** [reply]
> This may can help. See #2529

Thanks for the reference. It appears your work is related to indexing, not "keying"? (I'm not too familiar with Mongo terms). I suppose your PR might be a complement to this feature request, but I don't think it is the answer.
